### ISSUE TITLE
AirTerminal:SingleDuct:VAV:Reheat:VariableSpeedFan has heating control problem

### ISSUE THREAD

**Issue by EnergyArchmage** [NREL/EnergyPlus#5592]
## Issue overview

During work on a separate project (to add new fan) found that AirTerminal:SingleDuct:VAV:Reheat:VariableSpeedFan has control problems during heating. 

The example file called 5ZoneSupRetPlenVSATU.idf shows the problem.  During the morining hours of the Jan 14 day, there are some instances where the air flow rate toggles up to the maximum.  The model thinks it needs cooling and runs at the cooling max air flow when it should be staying at the minimum flow. 
### Details

The problem (found during debugging) is that the following is logic is not sufficent when the zone is in setback (13C) and the incoming supply air is already warmer (18C) than setpoint.  The cooling loads are positive but higher than the actuall heating load needed so the if block is true even where there is no cooling needed. 

```
    // Active cooling
    if ( QTotLoad < QCoolFanOnMin - SmallLoad && SysInlet( SysNum ).AirMassFlowRateMaxAvail > 0.0 && ! CurDeadBandOrSetback( ZoneNum ) ) {
```

Expanding the logic to be sure that QTotLoad is also negative appears to solve the problem. 

```
    if ( QTotLoad < ( -1.0 * SmallLoad ) && QTotLoad < (QCoolFanOnMin - SmallLoad) && SysInlet( SysNum ).AirMassFlowRateMaxAvail > 0.0 && ! CurDeadBandOrSetback( ZoneNum ) )
```

Some additional details for this issue (if relevant):
- Platform (Win8.1, )
- Version of EnergyPlus (i8.5)
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file is example/testfile 5ZoneSupRetPlenVSATU.idf
- [ ] Ticket added to Pivotal for defect 
- [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
I've long contended that CurDeadBandOrSetback should not ever be needed 
- that all equipment should be able to make a proper control decision 
  based  on remaining load to heating setpoint and remaining load to 
  cooling setpoint, but I haven't pursued going through every terminal 
  unit and system control to see if this is possible.

**Comment by mjwitte** [comment]
Closed via #6014
